services/operatorsinfo/operatorsinfo_inmemory_test.go

```diff
@@ -2,25 +2,115 @@ package operatorsinfo
 
 import (
 	"context"
+	"github.com/ethereum/go-ethereum/event"
 	"log/slog"
 	"math/big"
 	"os"
 	"reflect"
 	"testing"
 	"time"
 
-	"github.com/Layr-Labs/eigensdk-go/chainio/mocks"
+	apkregistrybindings "github.com/Layr-Labs/eigensdk-go/contracts/bindings/BLSApkRegistry"
+	blsapkreg "github.com/Layr-Labs/eigensdk-go/contracts/bindings/BLSApkRegistry"
+	regcoord "github.com/Layr-Labs/eigensdk-go/contracts/bindings/RegistryCoordinator"
 	"github.com/Layr-Labs/eigensdk-go/crypto/bls"
 	"github.com/Layr-Labs/eigensdk-go/logging"
 	"github.com/Layr-Labs/eigensdk-go/types"
 	"github.com/ethereum/go-ethereum/common"
 	gethtypes "github.com/ethereum/go-ethereum/core/types"
-	"go.uber.org/mock/gomock"
-
-	apkregistrybindings "github.com/Layr-Labs/eigensdk-go/contracts/bindings/BLSApkRegistry"
-	regcoordbindings "github.com/Layr-Labs/eigensdk-go/contracts/bindings/RegistryCoordinator"
 )
 
+type fakeAVSRegistryReader struct {
```

Comment:
@samlaf @eigen-vi my beef with at least this fake is, I am doing all these work for mimicking the same behavior what mocks would do. I am actually writing these fake to return what I what which is what mock does. Input to any function is not at all used (maybe that's what is wrong?) - all processing happens in constructor and then rest is just a read.

Reply:
sycned with @eigen-vi - convinced on fakes. I feel this stuff right now is not complex but eventually I think fakes would pay off.